test/functional/zmq_test.py

```diff
@@ -6,12 +6,23 @@
 import configparser
 import os
 import struct
+import time
 
 from test_framework.test_framework import BitcoinTestFramework, SkipTest
 from test_framework.util import (assert_equal,
                                  bytes_to_hex_str,
                                  )
 
+def wait_for_multipart(socket, timeout=60):
+    while True:
```

Comment:
This might be easier to read if made into a for loop:

    interval = 0.1
    for _ in range(int(timeout/interval)):
        try:
            return socket.recv_multipart(zmq.NOBLOCK)
        except zmq.error.Again:
            time.sleep(interval)
    raise AssertionError("Timed out waiting for zmq message")